cassandra-scheduler/src/test/java/io/mesosphere/mesos/frameworks/cassandra/PersistedCassandraClusterHealthCheckHistoryTest.java

```diff
@@ -0,0 +1,412 @@
+
+        hc2.setHealthy(true);
+        hc2.setMsg("msg");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(hc1.build(), hc2.build()));
+
+        hc2.clearMsg();
+        hc1.setMsg("msg");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(hc1.build(), hc2.build()));
+
+        hc1.clearMsg();
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(hc1.build(), hc2.build()));
+
+    }
+
+    @Test
+    public void testIsSimilarEntryNodeInfo() throws Exception {
+        CassandraFrameworkProtos.NodeInfo.Builder ni1 = CassandraFrameworkProtos.NodeInfo.newBuilder();
+        CassandraFrameworkProtos.NodeInfo.Builder ni2 = CassandraFrameworkProtos.NodeInfo.newBuilder();
+
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+        ni1.setUptimeMillis(1234);
+        ni2.setUptimeMillis(5678);
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.setClusterName("cluster");
+        ni2.setClusterName("cluster");
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.setNativeTransportRunning(true);
+        ni2.setNativeTransportRunning(true);
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni2.setNativeTransportRunning(false);
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni2.setNativeTransportRunning(true);
+        ni2.setClusterName("foo");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni2.setNativeTransportRunning(true);
+        ni2.setClusterName("cluster");
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.setDataCenter("dc");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.clearDataCenter();
+        ni2.setRack("rac");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+    }
+
+    @Test
+    public void testObjEquals() throws Exception {
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.objEquals(null, null));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals(null, ""));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals("", null));
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.objEquals("1", "1"));
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.objEquals(1, 1));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals("1", "2"));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals(1, 2));
+    }
```

Comment:
Damn out-of-order protocols ;)

In this case - I've added some code to silently drop records that are older that the most-recent one. Since we're not building a descent monitoring tool it should be fine IMO to drop outdated health-check info.